cherrybomb-engine/src/lib.rs

```diff
@@ -24,27 +24,43 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
 pub async fn run(config: &Config) -> anyhow::Result<Value> {
     verbose_print(config, None, "Starting Cherrybomb...");
 
-    // Reading OAS file to string
     verbose_print(config, None, "Opening OAS file...");
-    let oas_file = match std::fs::read_to_string(&config.file) {
-        Ok(file) => file,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error reading OAS file: {}", e));
-        }
-    };
-
-    // Parsing OAS file to JSON
-    verbose_print(config, None, "Parsing OAS file...");
-    let oas_json: Value = match serde_json::from_str(&oas_file) {
-        Ok(json) => json,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error parsing OAS file: {}", e));
+    // Reading OAS file to string let oas_json: Value;
+
+    let oas_json: Value;
+    let oas: OAS3_1;
+    match config.file.extension() {
```

Comment:
This should be done with a let oas_json = match, not by declaring a mut variable in the function scope

Reply:
rewrote the function :) 
Should be good now